README.md

```diff
@@ -4,569 +4,1111 @@
+  public_url_prefix: "http://mydomain.com/uploads"
+)
 ```
-post = Post.find(123)
-post.image # => <Saviour::File>
-post.image(:thumb) # => <Saviour::File>
+
+The `local_prefix` is the base prefix under which the storage will store files in the
+machine. You need to configure this accordingly to your use case and deployment strategies, for example, for rails
+and capistrano with default settings you'll have to store the files under `Rails.root.join("public/system")`,
+as this is by default the shared directory between deployments.
+
+The `public_url_prefix` is the base prefix to build the public endpoint from which you'll serve the assets.
+Same as before, you'll need to configure this accordingly to your deployment specifics.
+
+You can also assign a Proc instead of a String to dynamically calculate the value, useful when you have multiple
+asset hosts:
+
+`public_url_prefix: -> { https://media-#{rand(4)}.mywebsite.com/system/uploads/" }`
+
+This storage will take care of removing folders after they become empty.
+
+The optional extra argument `permissions` will allow you to set what permissions the files should have locally.
+This value defaults to '0644' and can be changed when creating the storage instance:
+
+```ruby
+Saviour::Config.storage = Saviour::LocalStorage.new(
+  local_prefix: Rails.root.join('public/system/uploads/'),
+  public_url_prefix: "http://mydomain.com/uploads",
+  permissions: '0600'
+)
 ```
 
-Finally, a couple of convenient methods are also added to the ActiveRecord object that just delegate to the `File` object:
+#### S3 Storage
 
+This storage will store files on Amazon S3, using the `aws-sdk-s3` gem. Example:
+
+```ruby
+Saviour::Config.storage = Saviour::S3Storage.new(
+  bucket: "my-bucket-name",
+  aws_access_key_id: "stub",
+  aws_secret_access_key: "stub",
+  region: "my-region",
+  public_url_prefix: "https://s3-eu-west-1.amazonaws.com/my-bucket/"
+)
 ```
-post = Post.find(123)
-post.image = File.open("/my/image.jpg") # This is equivalent to post.image.assign(File.open(...))
-post.image_changed? # This is equivalent to post.image.changed?
+
+The first 4 options `bucket`, `aws_access_key_id`, `aws_secret_access_key` and `region` are required for the
+connection and usage of your s3 bucket.
+
+The `public_url_prefix` is the base prefix to build the public endpoint from which the files are available.
+Normally you'll set it as in the example provided, or you can also change it accordingly to any CDN you may be
+using.
+
+You can also assign a Proc instead of a String to dynamically calculate the value, useful when you have multiple
+asset hosts:
+
+`public_url_prefix: -> { https://media-#{rand(4)}.mywebsite.com/system/uploads/" }`
+
+The optional argument `create_options` can be given to establishing extra parameters to use when creating files. For
+example, if you're digesting all your paths you might want to set up a large cache control value so that the
```

Comment:
`digesting` -> `adding a cache-buster/hash`? otherwise looks like someone is eating them